sqlbp/client.go

```diff
@@ -0,0 +1,190 @@
+package sqlbp
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"time"
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/jmoiron/sqlx"
+	"github.com/luna-duclos/instrumentedsql"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type tracer struct {
+	prefix string
+}
+
+type span struct {
+	tracer
+	parent opentracing.Span
+	err    error
+}
+
+// NewTracer returns a tracer that will fetch spans using opentracing's SpanFromContext function.
+func NewTracer(prefix string) instrumentedsql.Tracer {
+	return tracer{
+		prefix: prefix,
+	}
+}
+
+// GetSpan returns a span from the context, or creates one.
+func (t tracer) GetSpan(ctx context.Context) instrumentedsql.Span {
+
+	if ctx == nil {
+		parent := opentracing.StartSpan(
+			t.prefix,
+			tracing.SpanTypeOption{
+				Type: tracing.SpanTypeClient,
+			})
+
+		return &span{
+			parent: parent,
+			tracer: t,
+		}
+	}
+
+	return &span{parent: opentracing.SpanFromContext(ctx), tracer: t}
+}
+
+func (s *span) NewChild(name string) instrumentedsql.Span {
+	name = s.tracer.prefix + name
+	// s.parent == nil when db.query is used instead of db.queryContext
+	if s.parent == nil {
+		return &span{parent: opentracing.StartSpan(name), tracer: s.tracer}
+	}
+
+	return &span{
+		parent: s.parent.Tracer().StartSpan(
+			name,
+			opentracing.ChildOf(s.parent.Context()),
+			tracing.SpanTypeOption{
+				Type: tracing.SpanTypeClient,
+			},
+		),
+		tracer: s.tracer,
+	}
+}
+
+func (s *span) SetLabel(k, v string) {
+	s.parent.SetTag(k, v)
+}
+
+func (s *span) SetError(err error) {
+	if err == nil || err == driver.ErrSkip {
+		return
+	}
+	s.err = err
+}
+
+func (s *span) Finish() {
+	s.parent.FinishWithOptions(tracing.FinishOptions{
```

Comment:
the name `parent` starts to confuse me. why do you call `parent.FinishWithOptions` in `Finish` implementation? does `parent` actually mean this span, instead of its parent span?